python/oneseismic/scan/scan.py

```diff
@@ -260,22 +313,30 @@ def scan(stream, action):
     chunk = stream.read(header_size)
     header = segyio.field.Field(buf = chunk, kind = 'trace')
     action.add(header)
-    tracelen = action.tracelen()
+
+    tracelen  = action.tracelen()
+    tracesize = header_size + tracelen
+    samples   = action.observed['samples']
+
     stream.seek(tracelen, io.SEEK_CUR)
 
     trace_count = 1
     while True:
-        chunk = stream.read(header_size)
+        chunk = stream.read(tracesize)
         if len(chunk) == 0:
             break
 
-        if len(chunk) != header_size:
+        if len(chunk) != tracesize :
             msg = 'file truncated at trace {}'.format(trace_count)
             raise RuntimeError(msg)
 
-        header = segyio.field.Field(buf = chunk, kind = 'trace')
+        header = segyio.field.Field(buf = chunk[:header_size], kind = 'trace')
         action.add(header)
-        stream.seek(tracelen, io.SEEK_CUR)
+
+        trace = np.ndarray(samples, 'f4', buffer=chunk[header_size:])
```

Comment:
Micro nit, but I think you can drop either copy here. Don't re-open the patch for it, you can do `frombuffer(chunk[:range])` and either update it in-place, or let tonative() do the copy.